### ISSUE TITLE
Compatibility issue with Spring-boot 3.2

### ISSUE THREAD

**Issue by augustomelo** [apache/incubator-kie#5745]
I am opening this issue so the others could be aware when upgrading to spring-boot 3.2.

On spring-boot, it was added [nested jar support](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes#nested-jar-support), it seems that this causes problem when Kie needs to find kmodule.xml

logs of the problem:

```
2024-02-26T15:37:23.303Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Found kmodule: jar:nested:/srv/app.jar/!BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar!/META-INF/kmodule.xml
2024-02-26T15:37:23.619Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Unable to find pom.properties in /srv/app.jar/!BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar
2024-02-26T15:37:23.621Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : As folder project tried to fall back to pom.xml, but could not find one
2024-02-26T15:37:23.621Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Unable to load pom.properties from/srv/app.jar/!BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar
2024-02-26T15:37:23.621Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Cannot find maven pom properties for this project. Using the container's default ReleaseId
2024-02-26T15:37:23.623Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.i.InternalKieModuleProvider    : Creating KieModule for artifact org.default:artifact:1.0.0
2024-02-26T15:37:23.645Z ERROR 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Unable to build index of kmodule.xml url=jar:nested:/srv/app.jar/!BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar!/META-INF/kmodule.xml Range [57, 56) out of bounds for length 56
2024-02-26T15:37:23.652Z ERROR 1 --- [microservice-x] [  XNIO-1 task-2] x.d.ExceptionHandler    : RuntimeException java.lang.RuntimeException: Cannot find a default KieSession
```

after adding the option bellow on `build.gradle` 

```gradle
bootJar {
  loaderImplementation = org.springframework.boot.loader.tools.LoaderImplementation.CLASSIC
}
```

logs after applying the `LoaderImplementation.CLASSIC`
```
2024-02-26T15:48:52.940Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Found kmodule: jar:file:/srv/app.jar!/BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar!/META-INF/kmodule.xml
2024-02-26T15:48:53.269Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Unable to load pom.properties from/srv/app.jar!/BOOT-INF/lib/x-module-0.0.1-SNAPSHOT-plain.jar
2024-02-26T15:48:53.269Z  WARN 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.impl.ClasspathKieProject       : Cannot find maven pom properties for this project. Using the container's default ReleaseId
2024-02-26T15:48:53.271Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.k.b.i.InternalKieModuleProvider    : Creating KieModule for artifact org.default:artifact:1.0.0
2024-02-26T15:48:53.285Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.kie.builder.impl.KieContainerImpl  : Start creation of KieBase: defaultKieBase
2024-02-26T15:48:54.647Z  INFO 1 --- [microservice-x] [  XNIO-1 task-2] o.d.c.kie.builder.impl.KieContainerImpl  : End creation of KieBase: defaultKieBase
```

**Comment by mariofusco** [comment]
@augustomelo If you already have a project reproducing this issue can you please share it, so I could investigate from there? This will save me a bit of time and also ensure that we're looking at the same problem, thanks.

**Comment by augustomelo** [comment]
@mariofusco  bellow you may find the example project ( it is not the same, but it has the same error)

[project with error](https://github.com/augustomelo/kie-drool-problem)

logs from the the project:

```
2024-02-27T14:16:21.303Z  INFO 1 --- [           main] com.example.demo.DemoApplication         : Starting DemoApplication v0.0.1-SNAPSHOT using Java 21.0.2 with PID 1 (/srv/app.jar started by root in /)
2024-02-27T14:16:21.306Z  INFO 1 --- [           main] com.example.demo.DemoApplication         : No active profile set, falling back to 1 default profile: "default"
2024-02-27T14:16:22.105Z  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port 8080 (http)
2024-02-27T14:16:22.115Z  INFO 1 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2024-02-27T14:16:22.116Z  INFO 1 --- [           main] o.apache.catalina.core.StandardEngine    : Starting Servlet engine: [Apache Tomcat/10.1.19]
2024-02-27T14:16:22.141Z  INFO 1 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2024-02-27T14:16:22.142Z  INFO 1 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 771 ms
2024-02-27T14:16:22.425Z  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port 8080 (http) with context path ''
2024-02-27T14:16:22.436Z  INFO 1 --- [           main] com.example.demo.DemoApplication         : Started DemoApplication in 1.551 seconds (process running for 1.997)
2024-02-27T14:16:29.919Z  INFO 1 --- [nio-8080-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2024-02-27T14:16:29.919Z  INFO 1 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2024-02-27T14:16:29.920Z  INFO 1 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 1 ms
2024-02-27T14:16:29.934Z  INFO 1 --- [nio-8080-exec-1] c.e.demo.controller.SimpleController     : Before validate
2024-02-27T14:16:29.949Z  INFO 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : Found kmodule: jar:nested:/srv/app.jar/!BOOT-INF/lib/validations-module-0.0.1-SNAPSHOT-plain.jar!/META-INF/kmodule.xml
2024-02-27T14:16:30.121Z  WARN 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : Unable to find pom.properties in /srv/app.jar/!BOOT-INF/lib/validations-module-0.0.1-SNAPSHOT-plain.jar
2024-02-27T14:16:30.121Z  WARN 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : As folder project tried to fall back to pom.xml, but could not find one
2024-02-27T14:16:30.122Z  WARN 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : Unable to load pom.properties from/srv/app.jar/!BOOT-INF/lib/validations-module-0.0.1-SNAPSHOT-plain.jar
2024-02-27T14:16:30.122Z  WARN 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : Cannot find maven pom properties for this project. Using the container's default ReleaseId
2024-02-27T14:16:30.123Z  INFO 1 --- [nio-8080-exec-1] o.d.c.k.b.i.InternalKieModuleProvider    : Creating KieModule for artifact org.default:artifact:1.0.0
2024-02-27T14:16:30.128Z ERROR 1 --- [nio-8080-exec-1] o.d.c.k.b.impl.ClasspathKieProject       : Unable to build index of kmodule.xml url=jar:nested:/srv/app.jar/!BOOT-INF/lib/validations-module-0.0.1-SNAPSHOT-plain.jar!/META-INF/kmodule.xml
Range [57, 56) out of bounds for length 56
2024-02-27T14:16:30.131Z ERROR 1 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: java.lang.RuntimeException: Cannot find a default KieSession] with root cause

java.lang.RuntimeException: Cannot find a default KieSession
```

**Comment by Lionel-Tao** [comment]
Can I know if any update for this issue, also encounter when upgrade springboot to 3.2.7 @mariofusco mariofusco

**Comment by tkobayas** [comment]
I'm looking at this issue. I'll keep you posted.

**Comment by tkobayas** [comment]
Fixed by https://github.com/apache/incubator-kie-drools/pull/6064

**Comment by Uttam92477** [comment]
Which Drools version will this be released??

**Comment by tkobayas** [comment]
It will likely be included in 10.1. We are working on 10.0 release now. Unfortunately, I cannot provide ETA, but hopefully, some time soon.
